fs-cache/src/memory_limited_storage.rs

```diff
@@ -0,0 +1,273 @@
+use std::fs::{self, File};
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use std::time::SystemTime;
+
+use data_error::{ArklibError, Result};
+use linked_hash_map::LinkedHashMap;
+
+pub struct MemoryLimitedStorage<K, V> {
```

Comment:
It seems that we can safely merge `Cache` and `MemoryLimitedStorage` into a single struct unless we find other use-case for `MemoryLimitedStorage` except caching.